kiwi/package_manager/yum.py

```diff
@@ -187,14 +190,25 @@ def process_delete_requests(self, force=False):
             raise KiwiRequestError(
                 'None of the requested packages to delete are installed'
             )
-        delete_options = ['--nodeps', '--allmatches', '--noscripts']
         self.cleanup_requests()
-        return Command.call(
-            [
-                'chroot', self.root_dir, 'rpm', '-e'
-            ] + delete_options + delete_items,
-            self.command_env
-        )
+        if force:
+            delete_options = ['--nodeps', '--allmatches', '--noscripts']
+            return Command.call(
+                [
+                    'chroot', self.root_dir, 'rpm', '-e'
+                ] + delete_options + delete_items,
+                self.command_env
+            )
+        else:
+            chroot_yum_args = self.root_bind.move_to_root(self.yum_args)
+            return Command.call(
+                [
+                    'chroot', self.root_dir, self._get_yum_binary_name()
+                ] + chroot_yum_args + self.custom_args + [
+                    'erase'
```

Comment:
You can use `yum autoremove` to do the same thing that `dnf remove` does by default (remove and remove unneeded deps). `dnf autoremove` functions in the same way.

Cf. https://www.mankier.com/8/yum-deprecated#autoremove
Cf. https://www.mankier.com/8/dnf#Commands-Auto_Remove_Command